### PR TITLE
ign -> gz Macro Migration : gz-common

### PR DESCRIPTION
**Pull request by methylDragon**
See: https://github.com/gazebo-tooling/release-tools/issues/737

### CODE REVIEW

File: Migration.md [hunk 1]
```diff
@@ -18,18 +18,32 @@ release will remove the deprecated code.
...
+   4. `IGN_SPEED_OF_LIGHT`
+   5. `IGN_SLEEP_S`, `IGN_SLEEP_US`, `IGN_SLEEP_MS`, `IGN_SLEEP_NS`
+   6. `IGN_SYSTEM_TIME`, `IGN_SYSTEM_TIME_S`, `IGN_SYSTEM_TIME_US`, `IGN_SYSTEM_TIME_MS`, `IGN_SYSTEM_TIME_NS`
+   7. `IGN_ASSERT`
+   8. `IGNITION_COMMON_TINYOBJLOADER_IMPLEMENTATION` (`src` local, hard-tocked)
+   9. `IGN_PROFILER_ENABLE`, `IGN_PROFILE_THREAD_NAME`, `IGN_PROFILE_LOG_TEXT`, `IGN_PROFILE_BEGIN`, `IGN_PROFILE_END`, `IGN_PROFILE_L`, `IGN_PROFILE`, `IGN_PROFILE_VALID`
+   10. `IGN_CREATE_SPEC_INTERFACE`
+   11. `IGN_DUMMY_PLUGIN_PATH`
+   12. `IGNITION_UNITTEST_SPECIALIZED_PLUGIN_ACCESS`
+
```
Comment (chapulina) [inline Migration.md:46]: Missing `IGN_ASSERT`

File: include/ignition/common/EnumIface.hh [hunk 2]
```diff
@@ -17,3 +17,6 @@
 
 #include <gz/common/EnumIface.hh>
 #include <ignition/common/config.hh>
+
+#define IGN_ENUM(name, enumType, begin, end, ...) \
+  GZ_ENUM(name, enumType, begin, end, ...)
```
Comment (chapulina) [inline include/ignition/common/EnumIface.hh:22]: I think the `...` need to be `__VA_ARGS__`

File: include/ignition/common/detail/SpecializedPlugin.hh [hunk 3]
```diff
@@ -25,7 +25,7 @@
 // This preprocessor token should only be used by the unittest that is
 // responsible for checking that the specialized routines are being used to
 // access specialized plugin interfaces.
-#ifdef IGNITION_UNITTEST_SPECIALIZED_PLUGIN_ACCESS
+#ifdef GZ_UNITTEST_SPECIALIZED_PLUGIN_ACCESS
```
Comment (chapulina) [inline include/ignition/common/detail/SpecializedPlugin.hh:28]: These are deprecated, can we leave them as `IGNITION`?

File: profiler/src/CMakeLists.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-option(IGN_PROFILER_REMOTERY "Enable the Remotery profiler" TRUE)
+option(GZ_PROFILER_REMOTERY "Enable the Remotery profiler" TRUE)
```
Comment (chapulina) [inline profiler/src/CMakeLists.txt:1]: It would be good to tick-tock this option, it's exposed to users

File: profiler/src/CMakeLists.txt [hunk 5]
```diff
@@ -57,11 +57,11 @@ ign_add_component(profiler SOURCES ${PROFILER_SRCS} GET_TARGET_NAME profiler_tar
 
 # Always enable profiler so that it's built, but make it
 # private so that downstream users can still disable profiling
-target_compile_definitions(${profiler_target} PRIVATE "IGN_PROFILER_ENABLE=1")
+target_compile_definitions(${profiler_target} PRIVATE "GZ_PROFILER_ENABLE=1")
```
Comment (chapulina) [inline profiler/src/CMakeLists.txt:60]: This compiler definition too, downstream projects use it, so we need to tick-tock

**Review by chapulina — CHANGES_REQUESTED** [review]

File: Migration.md (hunk 1 above)
Comment (methylDragon) [reply Migration.md:46]: It's 7, my bad for putting it in a weird place :sweat_smile:

File: include/ignition/common/EnumIface.hh (hunk 2 above)
Comment (methylDragon) [reply include/ignition/common/EnumIface.hh]: https://github.com/gazebosim/gz-common/pull/366/commits/4c8849120b890b333e4561cd50985e456c186b97

File: include/ignition/common/detail/SpecializedPlugin.hh (hunk 3 above)
Comment (methylDragon) [reply include/ignition/common/detail/SpecializedPlugin.hh]: https://github.com/gazebosim/gz-common/pull/366/commits/603bbc67fca8c93ea4677296c223b434a3277c96

Comment (methylDragon) [reply include/ignition/common/detail/SpecializedPlugin.hh]: Also fixed in gz-plugin

File: profiler/src/CMakeLists.txt (hunk 5 above)
Comment (methylDragon) [reply profiler/src/CMakeLists.txt:62]: https://github.com/gazebosim/gz-common/pull/366/commits/dcb9ca9654e4aeae7e60bcf21426085ff444f888

File: profiler/src/CMakeLists.txt (hunk 4 above)
Comment (methylDragon) [reply profiler/src/CMakeLists.txt:1]: https://github.com/gazebosim/gz-common/pull/366/commits/dcb9ca9654e4aeae7e60bcf21426085ff444f888

Comment (methylDragon) [reply profiler/src/CMakeLists.txt:1]: Ah, fair point

Comment (methylDragon) [reply profiler/src/CMakeLists.txt]: https://github.com/gazebosim/gz-common/pull/366/commits/8bd54a075d97c2e8d484a2224cc0dbe0df843e68

**Review by chapulina — APPROVED** [review]

**Comment by methylDragon** [comment]
@osrf-jenkins run tests please!

**Comment by methylDragon** [comment]
@osrf-jenkins run tests please!

**Pull request merged**